fem/bilinearform.hpp

```diff
@@ -676,8 +689,25 @@ class MixedBilinearForm : public Matrix
 
    void operator=(const double a) { *mat = a; }
```

Comment:
I'm a little concerned about methods like this, and the `SpMat` methods earlier, that will fail if partial assembly was chosen.  Certainly for the `SpMat` methods we could make the argument that responsible developers should never call them if they requested that matrices not be assembled but we have no control over this and it can easily lead to runtime errors.  I prefer compile time errors whenever possible.

I wonder what people's reaction would be if I suggested that we should have an abstract base class for bilinear forms which contained a minimal interface and then two or more derived classes for concrete implementations of fully assembled and partially assembled bilinear forms.  Each concrete class could then have separate APIs containing only those methods which could actually be implemented.

Reply:
Do you think this should be addressed in this PR or discussed in the meeting for another PR?

Reply:
A meeting would probably be best.